libs/utils/src/vec_map.rs

```diff
@@ -1,35 +1,64 @@
 use std::{alloc::Layout, cmp::Ordering, ops::RangeBounds};
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum VecMapOrdering {
+    Greater,
+    GreaterOrEqual,
+}
```

Comment:
I can add others as well, but since they are not used anywhere I chose to not add them

Reply:
This is enough for now.